Builder fails on stale worktrees from previous failed runs instead of auto-recovering

## Problem

When a previous shepherd attempt fails after creating a worktree but before making any commits, the next shepherd attempt hits the stale worktree and fails immediately — requiring manual cleanup (`git worktree remove` + `git branch -D`).

**Observed in**: Shepherd run for #2662 (first attempt failed, manual cleanup required before retry succeeded → PR #2676).

### Error output

```
[ERROR] builder subprocess exited with code 1: worktree exists (branch=feature/issue-2662, commits_ahead=0, uncommitted=none); remote branch missing; no PR; labels=[loom:curated, loom:building]
```

The builder subprocess exited in ~3 seconds without doing any work.

### Compounding factor: branch naming mismatch

The stale worktree had branch `feature/issue-2662` (old naming convention), while the current `worktree.sh` creates `issue-N` branches (changed in 45e425a). The reset-in-place logic may not handle this naming mismatch — the anchor check reported `branch=issue-2662` but the diagnostic found `branch=feature/issue-2662`.

## Expected Behavior

If a worktree exists with:
- 0 commits ahead of main
- No uncommitted changes
- No remote branch
- No PR

The builder should treat it as a clean slate — remove and recreate (or reset in place on the correct branch) — rather than failing.

## Relevant Code

- **Worktree creation**: `.loom/scripts/worktree.sh` (lines 289-346) — has reset-in-place logic but may not handle branch name mismatches
- **Builder diagnostics**: `loom-tools/src/loom_tools/shepherd/phases/builder.py` `_gather_diagnostics()` (lines 3327-3382)
- **Builder validation**: `loom-tools/src/loom_tools/validate_phase.py` `validate_builder()` (lines 608-831)
- **Stale cleanup**: `builder.py` `_cleanup_stale_worktree()` exists but didn't trigger in this case

## Suggested Fix

In `worktree.sh` or the Python worktree handler, when a stale worktree is detected (0 commits, no uncommitted, no remote, no PR):

1. If the branch name doesn't match the expected convention (`issue-N`), remove the worktree and branch entirely, then recreate fresh
2. If the branch name matches, the existing reset-in-place logic should suffice

Alternatively, the shepherd's builder phase could detect this pattern before launching the builder subprocess and do the cleanup proactively.

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -588,8 +588,11 @@ class BuilderPhase:
                         },
                     )
 
-            # Clean up stale worktree to avoid leaving orphans
-            if exit_code in (6, 7) and diag.get("worktree_exists"):
+            # Clean up stale worktree to avoid leaving orphans (issue #2685).
+            # Previously only cleaned up for exit codes 6/7 (MCP/low-output),
+            # but any non-zero exit can leave a stale worktree that blocks
+            # the next shepherd attempt.
+            if diag.get("worktree_exists"):
                 if self._is_stale_worktree(ctx.worktree_path):
                     self._cleanup_stale_worktree(ctx)
 
@@ -2934,7 +2937,7 @@ class BuilderPhase:
                 wt_status.returncode == 0 and wt_status.stdout.strip()
             )
             log_res = subprocess.run(
-                ["git", "-C", str(wt), "log", "--oneline", "main..HEAD"],
+                ["git", "-C", str(wt), "log", "--oneline", "origin/main..HEAD"],
                 capture_output=True, text=True, check=False,
             )
             commits_ahead = (
@@ -3019,7 +3022,7 @@ class BuilderPhase:
             return None
 
         log_res = subprocess.run(
-            ["git", "-C", str(wt), "log", "--format=%s", "main..HEAD"],
+            ["git", "-C", str(wt), "log", "--format=%s", "origin/main..HEAD"],
             capture_output=True, text=True, check=False,
         )
         if log_res.returncode != 0 or not log_res.stdout.strip():
@@ -4114,12 +4117,45 @@ class BuilderPhase:
         and recreating the worktree, we reset it to origin/main and let the
         builder continue fresh.
 
+        If the branch name doesn't match the expected convention
+        (feature/issue-{N}), the worktree is removed entirely so
+        worktree.sh can recreate it with the correct branch name
+        (issue #2685).
+
         This approach:
         1. Preserves the worktree directory structure
         2. Ensures the branch is in sync with main
         3. Is faster than remove + recreate
         """
         if not ctx.worktree_path or not ctx.worktree_path.is_dir():
+            return
+
+        # Check for branch naming mismatch (issue #2685).
+        # If the stale worktree has the wrong branch name, reset-in-place
+        # would preserve the mismatch.  Remove entirely so worktree.sh
+        # recreates with the correct branch name.
+        expected_branch = f"feature/issue-{ctx.config.issue}"
+        branch_result = subprocess.run(
+            [
+                "git", "-C", str(ctx.worktree_path),
+                "rev-parse", "--abbrev-ref", "HEAD",
+            ],
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        actual_branch = (
+            branch_result.stdout.strip()
+            if branch_result.returncode == 0
+            else ""
+        )
+        if actual_branch and actual_branch != expected_branch:
+            log_warning(
+                f"Stale worktree branch mismatch: expected "
+                f"{expected_branch}, got {actual_branch} — removing "
+                f"worktree for clean recreation"
+            )
+            self._remove_stale_worktree(ctx)
             return
 
         # Fetch latest from origin to ensure we have current main

--- a/loom-tools/tests/shepherd/test_builder_escape_detection.py
+++ b/loom-tools/tests/shepherd/test_builder_escape_detection.py
@@ -209,7 +209,7 @@ class TestDetectWorktreeEscape:
                 side_effect=[
                     # git status --porcelain (worktree) - clean
                     _make_run_result(stdout=""),
-                    # git log --oneline main..HEAD (worktree) - no commits
+                    # git log --oneline origin/main..HEAD (worktree) - no commits
                     _make_run_result(stdout=""),
                 ],
             ),
@@ -239,7 +239,7 @@ class TestDetectWorktreeEscape:
                 side_effect=[
                     # git status --porcelain (worktree) - clean
                     _make_run_result(stdout=""),
-                    # git log --oneline main..HEAD (worktree) - has commits
+                    # git log --oneline origin/main..HEAD (worktree) - has commits
                     _make_run_result(stdout="abc1234 feat: implement feature\n"),
                 ],
             ),

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -935,6 +935,58 @@ class TestBuilderPhase:
         assert "exited with code 6" in result.message
         mock_cleanup.assert_called_once()
 
+    def test_generic_exit_code_stale_worktree_cleaned_up(
+        self, mock_context: MagicMock
+    ) -> None:
+        """Any non-zero exit with stale worktree should clean it up (issue #2685).
+
+        Previously only exit codes 6/7 triggered stale worktree cleanup.
+        Any non-zero exit can leave a stale worktree that blocks the next
+        shepherd attempt.
+        """
+        mock_context.check_shutdown.return_value = False
+        wt_mock = MagicMock()
+        wt_mock.is_dir.return_value = True
+        wt_mock.__bool__ = lambda self: True
+        mock_context.worktree_path = wt_mock
+
+        builder = BuilderPhase()
+        fake_diag = {
+            "summary": (
+                "worktree exists (branch=feature/issue-42, commits_ahead=0, "
+                "uncommitted=none); remote branch missing; no PR"
+            ),
+            "worktree_exists": True,
+            "has_uncommitted_changes": False,
+            "commits_ahead": 0,
+            "remote_branch_exists": False,
+            "pr_number": None,
+            "pr_has_review_label": False,
+        }
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.builder.get_pr_for_issue",
+                return_value=None,
+            ),
+            patch("loom_tools.shepherd.phases.builder.transition_issue_labels"),
+            patch(
+                "loom_tools.shepherd.phases.builder.run_phase_with_retry",
+                return_value=1,  # Generic non-zero exit (not 6 or 7)
+            ),
+            patch.object(builder, "_gather_diagnostics", return_value=fake_diag),
+            patch.object(builder, "_create_worktree_marker"),
+            patch.object(
+                builder, "_is_stale_worktree", side_effect=[False, True]
+            ),
+            patch.object(builder, "_cleanup_stale_worktree") as mock_cleanup,
+        ):
+            result = builder.run(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        assert "exited with code 1" in result.message
+        mock_cleanup.assert_called_once()
+
     def test_stuck_builder_includes_log_path(self, mock_context: MagicMock) -> None:
         """Builder stuck (exit 4) should include log file path in data."""
         mock_context.check_shutdown.return_value = False
@@ -9606,6 +9658,7 @@ class TestBuilderStaleWorktreeRecovery:
 
         # Create real mock context with real Path
         ctx = MagicMock()
+        ctx.config.issue = 42
         worktree = tmp_path / "issue-42"
         worktree.mkdir()
         ctx.worktree_path = worktree
@@ -9616,16 +9669,17 @@ class TestBuilderStaleWorktreeRecovery:
             ) as mock_run,
             patch.object(builder, "_remove_stale_worktree") as mock_remove,
         ):
-            # Fetch succeeds
-            # Reset succeeds
-            mock_run.return_value = MagicMock(returncode=0, stdout="", stderr="")
+            # Branch check returns correct branch, fetch and reset succeed
+            mock_run.return_value = MagicMock(
+                returncode=0, stdout="feature/issue-42\n", stderr=""
+            )
 
             builder._reset_stale_worktree(ctx)
 
             # Should not call remove since reset succeeded
             mock_remove.assert_not_called()
-            # Should have run fetch and reset
-            assert mock_run.call_count == 2
+            # Should have run: branch check, fetch, and reset
+            assert mock_run.call_count == 3
 
     def test_reset_stale_worktree_falls_back_to_remove(
         self, tmp_path: Path
@@ -9635,6 +9689,7 @@ class TestBuilderStaleWorktreeRecovery:
 
         # Create real mock context with real Path
         ctx = MagicMock()
+        ctx.config.issue = 42
         worktree = tmp_path / "issue-42"
         worktree.mkdir()
         ctx.worktree_path = worktree
@@ -9647,7 +9702,10 @@ class TestBuilderStaleWorktreeRecovery:
         ):
             def run_side_effect(cmd, **kwargs):
                 result = MagicMock()
-                if "fetch" in cmd:
+                if "rev-parse" in cmd:
+                    result.returncode = 0
+                    result.stdout = "feature/issue-42\n"
+                elif "fetch" in cmd:
                     result.returncode = 0
                 elif "reset" in cmd:
                     result.returncode = 1  # Reset fails
@@ -9661,6 +9719,42 @@ class TestBuilderStaleWorktreeRecovery:
 
             # Should call remove as fallback
             mock_remove.assert_called_once_with(ctx)
+
+    def test_reset_stale_worktree_branch_mismatch_removes(
+        self, tmp_path: Path
+    ) -> None:
+        """Stale worktree with wrong branch name is removed for clean recreation.
+
+        When a previous builder used a different branch naming convention
+        (e.g., issue-N vs feature/issue-N), reset-in-place would preserve the
+        mismatch.  The worktree should be fully removed so worktree.sh can
+        recreate with the correct branch name.  See issue #2685.
+        """
+        builder = BuilderPhase()
+
+        ctx = MagicMock()
+        ctx.config.issue = 42
+        worktree = tmp_path / "issue-42"
+        worktree.mkdir()
+        ctx.worktree_path = worktree
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.builder.subprocess.run"
+            ) as mock_run,
+            patch.object(builder, "_remove_stale_worktree") as mock_remove,
+        ):
+            # Branch name doesn't match expected feature/issue-42
+            mock_run.return_value = MagicMock(
+                returncode=0, stdout="issue-42\n", stderr=""
+            )
+
+            builder._reset_stale_worktree(ctx)
+
+            # Should remove (not reset) due to branch name mismatch
+            mock_remove.assert_called_once_with(ctx)
+            # Should NOT have called fetch or reset (only branch check)
+            assert mock_run.call_count == 1
 
     def test_remove_stale_worktree_cleans_up(self, tmp_path: Path) -> None:
         """Remove stale worktree removes worktree, local branch, and remote branch."""

